esmvalcore/experimental/utils.py

```diff
@@ -64,21 +66,36 @@ def get_all_recipes(subdir: str = None) -> list:
 def get_recipe(name: str) -> 'RecipeInfo':
     """Get a recipe by its name.
 
+    The function looks first in the local directory, and second in the
+    repository defined by the diagnostic path. The recipe name can be
+    specified with or without extension. The first match will be returned.
+
     Parameters
     ----------
-    name : str
-        Name of the recipe, i.e. ``examples/recipe_python.yml``
+    name : str, pathlike
+        Name of the recipe file, i.e. ``examples/recipe_python.yml``
 
     Returns
     -------
     RecipeInfo
+        Instance of :obj:`RecipeInfo` which can be used to inspect and run
+        the recipe.
+
+    Raises
+    ------
+    FileNotFoundError
+        If the name cannot be resolved to a recipe file.
     """
     locations = Path(), Path(DIAGNOSTICS_PATH, 'recipes')
-    filenames = name, name + '.yml'
+
+    if isinstance(name, Path):
+        filenames = (name, )
+    else:
+        filenames = (name, name + '.yml')
 
     for location in locations:
         for filename in filenames:
-            try_path = Path(location, filename)
+            try_path = Path(location, filename).expanduser()
             if try_path.exists():
```

Comment:
Perhaps it would be nice to print here something like "Found recipe_blablalba.yml" in "interesting/path". In that way, we could avoid any confusion as to which recipe is being used if there happen to be two of them when only 1 is expected or vice versa.

Reply:
Good idea, I added the search locations to the `FileNotFoundError`.